Sources/KituraWebSocket/WebSocketConnection.swift

```diff
@@ -105,9 +105,13 @@ extension WebSocketConnection: ChannelInboundHandler {
                 }
                 
                 if frame.fin {
-                    var data = unmaskedData(frame: frame)
-                    let text = data.readString(length: data.readableBytes) ?? ""
-                    fireReceivedString(message: text)
+                    let data = unmaskedData(frame: frame)
+                    if let text = data.getString(at: 0, length: data.readableBytes, encoding: .utf8) {
+                        fireReceivedString(message: text)
+                    } else {
+                        connectionClosed(reason: .dataInconsistentWithMessage, description: "Failed to convert received payload to UTF-8 String")
```

Comment:
The Kitura-WebSocket tests expect this error code to be 1005 i.e `WebSocketErrorCode(codeNumber: 1005)`